internal/scraper/clientFactory.go

```diff
@@ -0,0 +1,23 @@
+package scraper
+
+import (
+	"context"
+
+	discoveryv1 "github.com/syncromatics/kafmesh/internal/protos/kafmesh/discovery/v1"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+// ClientFactory generates DiscoveryClients for a pod
+type ClientFactory struct{}
+
+// Client gets a Discovery client for an address
+func (f *ClientFactory) Client(ctx context.Context, url string) (DiscoveryClient, func() error, error) {
```

Comment:
I was going to comment that this seems outside of typical Go factory patterns. However, in use it seems you've used this pattern to allow implementing services that use this to be mocked more easily. Is that correct?

Reply:
Correct. This was done to make unit testing the scraper job easier.